### PR TITLE
Command/draft

### LINKED ISSUE #7: Command draft
Details #1

### PR DESCRIPTION
**Pull request by Eomm**
Closes #7 

I implemented this first draft command in a straightforward way, I hope that the code is self-explaining.

For now, I'm running the script with:
```
node lib/cli.js draft --path '../fastify_1.x/' --tag v1.*
```

That print:
```
🐛 Fixed:

+ fix/mutated-schemas at startup (#1572)


🔍 Others:

+ test (#123)
+ test no PR in message
```

What is missing:

- [x] manage if there isn't a previous tag
- [x] logger for debug
- [ ] config for labels
- [x] docs
- [x] test: could you suggest me how to do tests? I'm running `git` commands so should I use proxyquire?

Any feedback or suggestions are welcome!

### CODE REVIEW

File: lib/release-message.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    } else if (labels.includes('semver-major')) {
+      semverBuckets.brakingChanges.push(message)
+    } else {
+      semverBuckets.mixed.push(message)
+    }
+  }
+
+  function toString () {
+    return printChangelog`🌟 Breaking changes:
+${semverBuckets.brakingChanges}
```
Comment (mcollina) [inline lib/release-message.js]: breaking

File: lib/release-message.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+${semverBuckets.bugfix}
+
+🔍 Others:
+${semverBuckets.mixed}`
+  }
+
+  function printChangelog (headers, ...changeList) {
+    let out = ''
+
+    for (const head in headers) {
```
[the remark this answers is not in the document]
Comment (Eomm) [reply lib/release-message.js]: Just to know: is only for clarity or there are other reasons?

[the remark this answers is not in the document]
Comment (mcollina) [reply lib/release-message.js]: This is going to traverse also up in the prototype chain. So you'll need to add `if (headers.hasOwnProperty(head))` inside.

File: lib/release-message.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+🚀 Features:
+${semverBuckets.feature}
+
+🐛 Fixed:
+${semverBuckets.bugfix}
+
+🔍 Others:
+${semverBuckets.mixed}`
+  }
```
[the remark this answers is not in the document]
Comment (Eomm) [reply lib/release-message.js]: For this kind of config (as per labels) would be better define a dotfile (like `.releasify`) or is better adding an argument with a template file for now?

[the remark this answers is not in the document]
Comment (mcollina) [reply lib/release-message.js]: I would just list all the PRs in a list for now, and do that in another PR.
I think a dotfile would be nice, yes.

File: lib/release-message.js [hunk 4]
```diff
@@ -2,6 +2,7 @@
 
 module.exports = function buildReleaseMessage () {
   const semverBuckets = {
+    list: [],
```
Comment (mcollina) [inline lib/release-message.js]: Do we need all the arrays afterwards?

File: lib/release-message.js [hunk 5]
```diff
@@ -19,35 +20,27 @@ module.exports = function buildReleaseMessage () {
...
+
+    // if (labels.includes('bugfix')) {
+    //   semverBuckets.bugfix.push(message)
+    // } else if (labels.includes('semver-minor')) {
+    //   semverBuckets.feature.push(message)
+    // } else if (labels.includes('semver-major')) {
+    //   semverBuckets.breakingChanges.push(message)
+    // } else {
+    //   semverBuckets.mixed.push(message)
+    // }
```
Comment (mcollina) [inline lib/release-message.js]: Please remove the commented code, git will retain in.

File: lib/release-message.js [hunk 6]
```diff
@@ -19,35 +20,27 @@ module.exports = function buildReleaseMessage () {
...
-
-🚀 Features:
-${semverBuckets.feature}
-
-🐛 Fixed:
-${semverBuckets.bugfix}
-
-🔍 Others:
-${semverBuckets.mixed}`
+    return printChangelog`📚 PR: ${semverBuckets.list}`
```
Comment (mcollina) [inline lib/release-message.js]: I think we should not use a tagged template here, mainly because I would like to load the template.

**Comment by Eomm** [comment]
> I’d love to have some tests, but the code is ok!

Sure!
I would work a little bit more to let the `publish` command quite "free" and define a better IN/OUT of the commands

File: lib/cli.js [hunk 7]
```diff
@@ -1,16 +1,24 @@
...
   const fn = require(`./commands/${command}`)
-  commist.register(command, (args) => {
+  commist.register(command, async (args) => {
     const opts = parseArgs(args)
     if (!needToShowHelp(command, opts)) {
-      fn(opts)
+      try {
+        const out = await fn(opts)
+        if (out) {
+          logger.info(out)
```
Comment (Eomm) [inline lib/cli.js]: This is where I'm not convinced: the out object is [this json](https://github.com/fastify/releasify/pull/8/files#diff-e3eecd52b8e957acc4df7269ea191a94R47).

Should I point that we log `out.message`?

**Comment by Eomm** [comment]
I think to have reached quite a good point now.

Not so much satisfied regarding the `test/helper.js . buildProxyCommand`.. for sure it will be refactored for next commands test

Coverage:

```
---------------------|----------|----------|----------|----------|-------------------|
File                 |  % Stmts | % Branch |  % Funcs |  % Lines | Uncovered Line #s |
---------------------|----------|----------|----------|----------|-------------------|
All files            |    97.18 |       88 |    90.63 |     97.1 |                   |
 lib                 |    95.18 |    82.35 |      100 |    95.18 |                   |
  args.js            |      100 |      100 |      100 |      100 |                   |
  cli.js             |    82.35 |    66.67 |      100 |    82.35 |          17,20,28 |
  git-directory.js   |    94.74 |       50 |      100 |    94.74 |                20 |
  github.js          |      100 |      100 |      100 |      100 |                   |
  man.js             |      100 |      100 |      100 |      100 |                   |
  release-builder.js |      100 |      100 |      100 |      100 |                   |
  validation.js      |      100 |      100 |      100 |      100 |                   |
 lib/commands        |      100 |      100 |    72.73 |      100 |                   |
  draft.js           |      100 |      100 |      100 |      100 |                   |
  help.js            |      100 |      100 |      100 |      100 |                   |
  login.js           |      100 |      100 |        0 |      100 |                   |
  publish.js         |      100 |      100 |        0 |      100 |                   |
  whoami.js          |      100 |      100 |        0 |      100 |                   |
---------------------|----------|----------|----------|----------|-------------------|
```

Badges and other stuff in another PR

@mcollina If you spot other messy thing I'll take care of it 💪

File: README.md [hunk 8]
```diff
@@ -1,2 +1,46 @@
...
+
+Print out the new version that should be released with its changelog listing the commit messages.
+
+The commits shown are
++ from: the `HEAD` of your local project 
++ to: the _first commit_ or the _last version tag_ you created
+
+Moreover, if the commit message is written with the pattern: `text describing (#123)`, where
+`(#123)` is the pull request ID, the labels of that PR are downloaded and processed by the template
+engine. These info are fetched from GitHub, keep in mind that there are [rete limits](https://developer.github.com/v3/#rate-limiting).
```
Comment (mcollina) [inline README.md]: rete -> rate

**Pull request merged**